### DISCUSSION TITLE
Refresh/Download filters after start

### DISCUSSION THREAD

**Discussion by Cebeerre** [AdguardTeam/AdGuardHome#3365] [Ideas]
Hi, 

I've my /data directory in a tmpfs as I've AGH running on top of openwrt x86_64 that is sitting on an SD card. After a router restart, I obviously loose the statistics and the txt files that contain the filters. This makes AGH "just work" without filters till the next "Filter updates inverval" happens as the "Rule Count" is 0. 

Is there any way to force a filter update after restart ? Sounds reasonable implementing something like this ? I'm aware of my specific setup, but double checking that the filter exists makes sense to me ... starting without rules when there are filters defined and force an update, looks like a good safety check ...

Thanks !

**Comment by ameshkov** [comment]
Please check the API docs: https://github.com/AdguardTeam/AdGuardHome/tree/master/openapi

You should be able to trigger filters updates using `/control/filtering/refresh`

**Reply by Cebeerre** [reply]
> Please check the API docs: https://github.com/AdguardTeam/AdGuardHome/tree/master/openapi
> 
> You should be able to trigger filters updates using `/control/filtering/refresh`

Yeah, I actually have a script that does this but I thought it makes no sense to have a filter with 0 rules and that AGH starts anyway because it doesn't actually check that the .txt in the filesystem is in sync with the online one ... it maybe doesn't hurt have to have a config flag jointly with "Filter update interval" like "Update after start" or something like that ...
